st2api/tests/unit/controllers/v1/test_actions.py

```diff
@@ -809,6 +843,165 @@ def test_delete_action_belonging_to_system_pack(self):
         del_resp = self.__do_delete(action_id, expect_errors=True)
         self.assertEqual(del_resp.status_int, 400)
 
+    @mock.patch.object(os.path, "isdir", mock.MagicMock(return_value=True))
+    @mock.patch("st2api.controllers.v1.actions.clone_action")
+    @mock.patch.object(
+        action_validator, "validate_action", mock.MagicMock(return_value=True)
+    )
+    def test_clone(self, mock_clone_action):
+        source_post_resp = self.__do_post(ACTION_16)
+        self.assertEqual(source_post_resp.status_int, 201)
+        dest_post_resp = self.__do_post(ACTION_17)
+        self.assertEqual(dest_post_resp.status_int, 201)
+        dest_data_body = {
+            "dest_pack": ACTION_17["pack"],
+            "dest_action": "clone_action_2",
+        }
+        source_ref_or_id = "%s.%s" % (ACTION_16["pack"], ACTION_16["name"])
+        clone_resp = self.__do_clone(dest_data_body, source_ref_or_id)
+        self.assertEqual(clone_resp.status_int, 201)
+        self.__do_delete(self.__get_action_id(source_post_resp))
+        self.__do_delete(self.__get_action_id(dest_post_resp))
+        self.__do_delete(self.__get_action_id(clone_resp))
+
+    @mock.patch.object(os.path, "isdir", mock.MagicMock(return_value=True))
+    @mock.patch("st2api.controllers.v1.actions.clone_action")
+    @mock.patch.object(
+        action_validator, "validate_action", mock.MagicMock(return_value=True)
+    )
+    def test_clone_overwrite(self, mock_clone_action):
+        source_post_resp = self.__do_post(ACTION_16)
+        self.assertEqual(source_post_resp.status_int, 201)
+        dest_post_resp = self.__do_post(ACTION_17)
+        self.assertEqual(dest_post_resp.status_int, 201)
```

Comment:
Create an ACTION_17 with something different so we can test below that we overwrote it with something else.